### DISCUSSION TITLE
Is it normal that DAOs skip inserting default values?

### DISCUSSION THREAD

**Discussion by kaqqao** [jOOQ/jOOQ#15226] [Q&A] [answered]
I have a PostgreSQL table similar to:

```sql
create table item (
    id uuid default gen_random_uuid() primary key not null,
    name varchar,
    created_at timestamptz not null default current_timestamp,
);
```

If I use the generated DAO to insert an item `dao.insert(item)`, I get a non-null constraint violation for `created_at`.

I see a related issue https://github.com/jOOQ/jOOQ/issues/10386 is closed as _won't fix_, but that seems to be talking exclusively about blank inserts (with nothing but default values). But I may be misunderstanding it.

Is what I'm seeing the expected behavior? Anything I can do to make the default timestamp apply?

**Comment by lukaseder** [comment, marked as answer]
Thanks for your message and for your patience. The discussions don't seem to be part of our "open questions" report yet, which is why this hasn't been answered yet.

If jOOQ knows about both your `NOT NULL` constraint as well as the `DEFAULT` expression, then there shouldn't be any explicit `NULL` values in the `INSERT` statement. Possibly a bug. May I ask you to report one here, along with a complete reproducer? https://www.jooq.org/doc/latest/manual/sql-building/model-api/model-api-replacement/
